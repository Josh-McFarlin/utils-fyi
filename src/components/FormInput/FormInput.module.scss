.root {
  width: 100%;
  color: #333333;
  box-sizing: border-box;
  padding: 8px;
}
