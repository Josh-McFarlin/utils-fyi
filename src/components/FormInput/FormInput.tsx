import * as React from "react";
import clsx from "clsx";
import classes from "./FormInput.module.scss";

interface Props extends React.HTMLProps<HTMLInputElement> {
  className?: string;
  inputClassName?: string;
  disabled?: boolean;
  label?: string;
  value: string | number;
  placeHolder?: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

/**
 * An input that can be used for any type of data.
 */
const FormInput = ({
  className,
  inputClassName,
  disabled = false,
  label,
  value,
  placeHolder,
  onChange,
  ...rest
}: Props) => (
  <label className={clsx(classes.root, className)}>
    {label != null && <p>{label}</p>}
    <input
      className={inputClassName}
      disabled={disabled}
      placeholder={placeHolder}
      value={value}
      onChange={onChange}
      {...rest}
    />
  </label>
);

export default FormInput;
