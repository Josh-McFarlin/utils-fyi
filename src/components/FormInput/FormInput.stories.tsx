import * as React from "react";
import FormInput, { PropTypes } from "./FormInput";

export default {
  title: "FormInput",
  component: FormInput,
};

export const Component = (args: PropTypes) => {
  const [value, setValue] = React.useState("This is a FormInput!");

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    setValue(event.target.value);

  return <FormInput {...args} value={value} onChange={handleChange} />;
};

export const WithLabel = (args: PropTypes) => {
  const [value, setValue] = React.useState("");

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    setValue(event.target.value);

  return <FormInput {...args} value={value} onChange={handleChange} />;
};

WithLabel.args = { label: "Label" };

export const WithPlaceholder = (args: PropTypes) => {
  const [value, setValue] = React.useState("");

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    setValue(event.target.value);

  return <FormInput {...args} value={value} onChange={handleChange} />;
};

WithPlaceholder.args = { placeHolder: "Placeholder" };

export const WithEmoji = (args: PropTypes) => {
  const [value, setValue] = React.useState("Josh ðŸ˜Ž");

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    setValue(event.target.value);

  return <FormInput {...args} value={value} onChange={handleChange} />;
};

WithEmoji.args = { placeHolder: "Name ðŸ˜€" };
