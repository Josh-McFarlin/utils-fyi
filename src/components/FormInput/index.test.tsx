import React from "react";
import { render } from "../../../test/testUtils";
import FormInput from "./FormInput";

// eslint-disable-next-line @typescript-eslint/no-empty-function
const noOp = () => {};

describe("FormInput Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <FormInput value="This is a FormInput!" onChange={noOp} />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
