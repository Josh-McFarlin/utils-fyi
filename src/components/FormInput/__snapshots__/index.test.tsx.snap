// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component matches snapshot 1`] = `
<DocumentFragment>
  <label
    class="root"
  >
    <input
      value="This is a FormInput!"
    />
  </label>
</DocumentFragment>
`;
